src/java/org/apache/cassandra/service/DefaultFSErrorHandler.java

```diff
@@ -60,6 +60,7 @@ public void handleFSError(FSError e)
 
         switch (DatabaseDescriptor.getDiskFailurePolicy())
         {
+            case die:
```

Comment:
the order of these enums in this switch is different from the order in `handleStartupFSError`

Reply:
Is not it true that when policy is `die` and we reach this method, instead of killing the jvm, you will only stop transports? There is this in `cassandra.yaml`:

````
# die
#   shut down gossip and client transports and kill the JVM for any fs errors or
#   single-sstable errors, so the node can be replaced.
````

As I read it, we should just kill whole JVM, no?

Reply:
Also, do not we miss `die` policy in `handleCorruptSSTable` method in this class? There is only `stop_paranoid`.

Reply:
https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/utils/JVMStabilityInspector.java#L142

@smiklosovic I think at this step we will kill jvm when policy is die with FSerror or CorruptSSTableException

Reply:
ah right ...

Reply:
@Runtian can you actually explain where what does this change do? Because to my understanding it will just stop transports? But the objective is to kill the whole jvm? What are we stopping the transports for when we kill the jvm anyway?

Reply:
Right, it is here to not reach the default so IllegalStateException will not be thrown.

We can also just add die with ignore to do nothing, but I think it is better to add die with stop and stop_paranoid so it tries to stop the transport and gossip etc. Also added the similar change to `handleCorruptSSTable`